sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -63,8 +154,107 @@ export interface AssessmentSentiment {
     text: string;
 }
 
+// @public
+export type Association = "subject" | "other";
+
 export { AzureKeyCredential }
 
+// @public
+export interface BatchActionErrorResult extends BatchActionState {
+    readonly error: TextAnalysisError;
+    readonly failedOn: Date;
+}
+
+// @public
+export type BatchActionResult<T> = BatchActionSuccessResult<T> | BatchActionErrorResult;
+
+// @public
+export interface BatchActionState {
+    readonly actionName?: string;
+    readonly statistics?: TextDocumentBatchStatistics;
+}
+
+// @public
+export interface BatchActionSuccessResult<T> extends BatchActionState {
+    readonly completedOn: Date;
+    readonly error?: undefined;
+    readonly results: T[];
+}
+
+// @public
+export interface BeginAnalyzeBatchOptions extends TextAnalysisOperationOptions {
+    displayName?: string;
+    resumeFrom?: string;
+    updateIntervalInMs?: number;
+}
+
+// @public
+export type Certainty = "positive" | "positivePossible" | "neutralPossible" | "negativePossible" | "negative";
+
+// @public
+export interface ClassificationCategory {
+    category: string;
+    confidenceScore: number;
+}
+
+// @public
+export type Conditionality = "hypothetical" | "conditional";
+
+// @public
+export interface CreateAnalyzeBatchPollerOptions extends TextAnalysisOperationOptions {
+    updateIntervalInMs?: number;
+}
+
+// @public
+export interface CustomActionMetadata {
+    readonly deploymentName: string;
+    readonly projectName: string;
+}
+
+// @public
+export type CustomEntityRecognitionAction = ActionCustom & {
+    stringIndexType?: StringIndexType;
+};
+
+// @public
+export type CustomEntityRecognitionErrorResult = TextAnalysisErrorResult;
+
+// @public
+export type CustomEntityRecognitionResult = CustomEntityRecognitionSuccessResult | CustomEntityRecognitionErrorResult;
+
+// @public
+export interface CustomEntityRecognitionSuccessResult extends TextAnalysisSuccessResult {
+    entities: Entity[];
+}
+
+// @public
+export type CustomMultiLabelClassificationAction = ActionCustom & {};
```

Comment:
I don't understand what the intersection is doing here. You could make this a named interface that extends ActionCustom or make this an alias for ActionCustom as you've done with other types here that don't need to add fields to "logical" supertypes.

Reply:
this is also auto-generated :(

Reply:
For the record, I opened https://github.com/Azure/autorest.typescript/pull/1386 to fix the intersection behavior.